DancingGoat/Utils/ProjectUtils.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace DancingGoat.Utils
+{
+    public class ProjectUtils
+    {
+        public static string GetProjectId()
+        {
+            var requestUrl = System.Web.HttpContext.Current.Request.Url;
+            var subdomain = requestUrl.GetSubdomain();
+            var decodedGuid = Guid.ParseExact(subdomain, "N");
+
+            if (decodedGuid == null)
+            {
+                throw new InvalidOperationException(nameof(decodedGuid));

```

Comment:
You didn't read the rest of my comment, ParseExact never returns null, as Guid can't be null